clients/application-amqp/src/main/java/org/eclipse/hono/application/client/amqp/ProtonBasedCommandSender.java

```diff
@@ -0,0 +1,100 @@
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.application.client.CommandSender;
+import org.eclipse.hono.client.HonoConnection;
+import org.eclipse.hono.client.SendMessageSampler;
+import org.eclipse.hono.client.StatusCodeMapper;
+import org.eclipse.hono.client.amqp.SenderCachingServiceClient;
+import org.eclipse.hono.util.AddressHelper;
+import org.eclipse.hono.util.CommandConstants;
+import org.eclipse.hono.util.MessageHelper;
+
+import io.opentracing.SpanContext;
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.proton.ProtonHelper;
+
+/**
+ * A vertx-proton based client for sending commands.
+ *
+ * @see <a href="https://www.eclipse.org/hono/docs/api/command-and-control/">
+ *      Command &amp; Control API for AMQP 1.0 Specification</a>
+ */
+public class ProtonBasedCommandSender extends SenderCachingServiceClient implements CommandSender {
+
+    /**
+     * Creates a new vertx-proton based command sender.
+     *
+     * @param connection The connection to the Hono server.
+     * @param samplerFactory The factory for creating samplers for tracing AMQP messages being sent.
+     * @throws NullPointerException if any of the parameters except the closeHandler is {@code null}.
+     */
+    public ProtonBasedCommandSender(
+            final HonoConnection connection,
+            final SendMessageSampler.Factory samplerFactory) {
+        super(connection, samplerFactory);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws NullPointerException if tenantId, deviceId, command, correlationId or replyId is {@code null}.
+     */
+    @Override
+    public Future<Void> sendAsyncCommand(final String tenantId, final String deviceId, final String command,
+            final String contentType, final Buffer data, final String correlationId, final String replyId,
+            final Map<String, Object> properties, final SpanContext context) {
+        Objects.requireNonNull(tenantId);
+        Objects.requireNonNull(deviceId);
+        Objects.requireNonNull(command);
+        Objects.requireNonNull(correlationId);
+        Objects.requireNonNull(replyId);
+
+        return getOrCreateSenderLink(CommandConstants.NORTHBOUND_COMMAND_REQUEST_ENDPOINT, tenantId)
+                .recover(thr -> Future.failedFuture(StatusCodeMapper.toServerError(thr)))
+                .compose(sender -> {
+                    final String targetAddress = AddressHelper
+                            .getTargetAddress(CommandConstants.NORTHBOUND_COMMAND_REQUEST_ENDPOINT, tenantId, deviceId,
+                                    connection.getConfig());
+                    final Message msg = createMessage(tenantId, deviceId, command, contentType, data, correlationId,
+                            replyId, targetAddress, properties);
+                    return sender.sendAndWaitForOutcome(msg, newChildSpan(context, "send Command " + command));
```

Comment:
Just use "send command" as span name here - the span name should be fixed (will occur in the "operation" dropdown in the Jaeger UI).

Reply:
The " + command" part seems to be still in there.

Reply:
Ah ok, now I see what you meant. Done